orttraining/orttraining/python/ort_trainer.py

```diff
@@ -728,6 +728,7 @@ def _init_onnx_model(self, inputs):
             return
 
         if self.torch_model_ is not None:
+            self.torch_model_.cpu()
```

Comment:
Should we cache the state and revert it back to original after the export has been done?

Reply:
the reverse is harder because the model can be originally on different devices. There isn't any api available to achieve that.

Reply:
Have you tried

```python
# save original device
orig_device=next(self.torch_model_.parameters()).device
# put it on cpu
self.torch_model_.cpu()
# restore to original device
self.torch_model_.to(orig_device)
```
?

Although it might be possible, I am not sure how useful it would be. As per my understading, the solely purpose of `self.torch_model_` is to be used for onnx conversion, so the original state doesnt look that important.

Reply:
There is one catch. The code above assume all layers are in the same device, which is not always true, as described [here](https://pytorch.org/tutorials/intermediate/model_parallel_tutorial.html)

For this case, we would have to iterate through `model.parameters()` and save each device for future restore, which is a lot more work

Reply:
yes, and there are also cached_buffer, and others.

And the restriction on cpu device is not from export, but from training. GPU memory needs to be reserved for onnx, that's why to set the model on CPU.